### PR TITLE
Routine Cargo Departmental Deliveries

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Document: https://hackmd.io/@bazelart/HkY-SO9VF

Each department's request console is upgraded. Instead of making requests, they are only able to see crates related to their department that helps the department. They can order one for free (so not out of anyone's wallet) and it will arrive at cargo. Cargo gets the price of the ordered crate for bringing it to the department that ordered it (via an area check, of which the crate will remain locked until satisfied, emitter cracking aside).

Ordering a crate puts the console on a cooldown depending on the price of the crate ordered. The time ranges from 10 minutes at the lowest value, to capped at 20 minutes at 4x the default crate's price. the price in time follows a ease in out circular function, where the time increases slowly as the time goes up, but a lot in the middle. towards the higher end it slows down again.

Cargo will start with far less money, but this is countered by the new source of income.

- [x] Finish Mapping it
- [x] Finish tgui panel for department orders
- [x] ~Testmerge it for balance and feedback~ who cares
- [x] ~Maybe give multiple destination areas, just in case one area gets obliterated~ who cares!

## Why It's Good For The Game

Cargo starts with a budget to spend on themselves, which lets them order their department rewards before doing anything. There is no inherent reward to delivering to other departments other than social expectations of filling the job, which means cargo will simply ignore deliveries and requests if there are more self-important things going on, since those deliveries both cost cargo and do not help cargo in any way

## Changelog

:cl:
add: replaced the request consoles in each department with department order consoles, which order for free on a cooldown. cargo gets these orders and delivers the crates, which are locked until delivery. upon delivery, cargo gets paid the value of the crate, and can then sell the crate back on the shuttle.
balance: cargo doesn't start with a budget, other departments get what their budget was split up amongst them
/:cl:

### CODE REVIEW

**Comment by TheBonded** [comment]
finally, we add a reason to use #cargobus channel. now we just need that to have runechat (or a ping that informs people of new messages)

**Comment by tralezab** [comment]
> does this edit the crate pool the departments have access to right now? mech pilot costume crates are an essential resource here in RnD

I agree that department themed costumes should be available but I'll have to reorder groupings to allow costume crates to be with certain groups

**Comment by MrDoomBringer** [comment]
fantastic idea 👍

**Comment by MrDoomBringer** [comment]
> I'm concerned about only being able to order two crates a round per department

The time scales on price, so they can order 6 to 3 a round for completely free. They can walk to cargo and order more if they need, and I believe that hallway traversing is the kind of thing we want to encourage anyways.

**Comment by Iamgoofball** [comment]
Oh okay I misunderstood

**Comment by MrDoomBringer** [comment]
hell yeah brother

File: code/__HELPERS/maths.dm [hunk 1]
```diff
@@ -123,3 +123,10 @@
 			return "centuple"
 		else //It gets too tedious to use latin prefixes from here.
 			return "[number]-tuple"
+
+///returns the position from 0 to 1 that y should be in, given x progress from 0 to 1 in the inbetweening.
```
Comment (optimumtact) [inline code/__HELPERS/maths.dm:127]: was this written when you were high? what does this do

**Comment by tralezab** [comment]
time to fix merge conflicts again :finnadie:

**Comment by MrDoomBringer** [comment]
huge

**Comment by NanoCats** [comment]
MAJOR ISSUES DETECTED during a round with this PR merged, (I also made an issue for the strictly bug part):

A: no contraband/broad setting (Boo!), deconstructing the console at all causes big issues 
B: only able to order things "relevant" to your department (let engineering order pizza damn you)
C: Clown can't order toys (or anything else) at service wtf
D: You still can't request things from cargo if you're an assistant or whatever and you don't have your own department. **put, a, fucking, cargo, request, console, in, the, cargo, lobby, _please_**

**Comment by optimumtact** [comment]
assistants  are irrelevant.

**Comment by adamkad1** [comment]
Assistants do plenty of bounties because they dont have any cash otherwise. Makes pretty good cash cows for cargo

**Comment by NanoCats** [comment]
> assistants are irrelevant.

There are plenty of reasons you might want to go order something directly from cargo which is why there should be a request console there.

including but not limited to wanting to speak with a person and ask them to do X with your delivery, also you can't request custom items from your department with this cargo rework. Expecting someone to be at the desk 24/7 is not a reasonable expectation either. 

As an engineer, I can easily build a new cargo console somewhere else, but other departments have no such luxury. 


also what the fuck does this do to the IRN app anyways I didn't get to test.

**Comment by adamkad1** [comment]
Can cargo get paid for each piece of mail opened too?

**Pull request merged**